src/Controls/src/Core/Handlers/Items/CarouselViewHandler.Android.cs

```diff
@@ -53,25 +54,40 @@ public override Size GetDesiredSize(double widthConstraint, double heightConstra
 		{
 			_widthConstraint = widthConstraint;
 			_heightConstraint = heightConstraint;
+
+			if (!double.IsInfinity(_widthConstraint))
+				_widthConstraint = Context.ToPixels(_widthConstraint);
+
+			if (!double.IsInfinity(_heightConstraint))
+				_heightConstraint = Context.ToPixels(_heightConstraint);
+
 			return base.GetDesiredSize(widthConstraint, heightConstraint);
 		}
 
-		int GetItemWidth()
+		public override void PlatformArrange(Rect frame)
+		{
+			_widthConstraint = Context.ToPixels(frame.Width);
```

Comment:
If the `GetDesiredSize` call was using infinite then these will set the contraints to the arrived at values